crypto/s2n_ecc_evp.c

```diff
@@ -62,12 +62,15 @@ const struct s2n_ecc_named_curve s2n_ecc_curve_x25519 = {
 };
 #endif
 
-const struct s2n_ecc_named_curve *const s2n_ecc_evp_supported_curves_list[] = {
+const static struct s2n_ecc_named_curve *const supported_curves_list[] = {
     &s2n_ecc_curve_secp256r1,
     &s2n_ecc_curve_secp384r1,
+#if MODERN_EC_SUPPORTED
+    &s2n_ecc_curve_x25519,
+#endif
```

Comment:
Should this list be managed in the same place as the preference lists, and should we have a way to verify it contains all curves on our preference lists?

Reply:
The reason I had the static list of all curves here is it is needed in the `s2n_ecc_evp_find_supported_curve` function, but with the addition of madeline's new changes `s2n_parse_client_supported_groups_list` and `s2n_choose_supported_group` replaces it, so we don't really use this function anymore.